doc/document/PortingToFreeRTOSAWS.md

```diff
@@ -0,0 +1,298 @@
+# FreeRTOS AWS porting guide
+
+## Introduction
+
+The FreeRTOS Cellular Library exposes the capability of cellular modems with AT command interface. This guide provides a method to integrate FreeRTOS Cellular Library with FreeRTOS AWS. We use windows simulator and BG96 cellular module as an example. FreeRTOS Cellular Library is sub-moduled in vendor’s porting folder. To communicate with cellular modem, an implementation of comm interface is required. There are some comm interface implementations can be referenced to reduced porting effort. An example implementation of secure sockets with FreeRTOS cellular APIs is also provided in this guide. FreeRTOS AWS demo application can make use of secure sockets cellular to connect to AWS IoT Core.
+
+
+> Windows simulator and FreeRTOS AWS 202012.00 are used in this example. User may need to adapt to different platform or FreeRTOS AWS version.
+
+> This document is provided as an example. User doesn't need to follow this porting method if the have their own implementation.
+
+## Porting steps
+
+1. Create the ports/cellular folder with cellular library and adapting functions
+2. Create the ports/comm_if folder with comm interface implementation
+3. Edit application board CMakeLists.txt to include cellular library
+4. Edit main.c to call cellular library APIs
+
+## Folder structure
+
+Windows simulator is used as an example. Only files or folders related to porting are listed.
+
+```
+./vendors/pc/boards/windows
+├── aws_demos
+│   └── application_code
+│       ├── cellular_setup.c
+│       └── main.c ( Reference the sample code below )
+├── CMakeLists.txt ( Reference the sample code below )
+└── ports
+    ├── cellular
+    │   ├── CMakeLists.txt ( Reference the sample code below )
+    │   ├── configs
+    │   │   └── cellular_config.h
+    │   ├── library ( submodule : FreeRTOS Cellular Library )
+    │   ├── platform
+    │   │   ├── cellular_platform.c
+    │   │   └── cellular_platform.h
+    │   └── secure_sockets
+    │       └── iot_secure_sockets.c
+    └── comm_if
+        └── comm_if_windows.c
+        
+```
+Reference implementation of these files
+* [cellular_setup.c](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Demo/blob/master/source/cellular_setup.c)
+* [cellular_config.h](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Demo/blob/main/source/cellular/bg96/cellular_config.h)
+* [cellular_platform.c](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Demo/blob/master/source/cellular/cellular_platform.c)
+* [cellular_platform.h](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Demo/blob/master/source/cellular/cellular_platform.h)
+* [iot_secure_sockets.c](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/blob/main/doc/samples/secure_sockets_cellular/iot_secure_sockets.c)
+* [comm_if_windows.c](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Demo/blob/main/source/cellular/comm_if_windows.c)
+
+## AFR module dependency

```

Comment:
I would suggest we use a block diagram to explain the dependency with a simplfied description.